src/modules/commander/Arming/PreFlightCheck/checks/preArmCheck.cpp

```diff
@@ -143,6 +149,19 @@ bool PreFlightCheck::preArmCheck(orb_advert_t *mavlink_log_pub, const vehicle_st
 		}
 	}
 
+	// This requires a valid takeodd waypoint to be set
+	uORB::Subscription setpoint_sub(ORB_ID(position_setpoint_triplet));
+	position_setpoint_triplet_s setpoint = {};
+	setpoint_sub.update(&setpoint);
```

Comment:
I have two concerns here:
1. The triplet is quite big and hence this needs CPU to copy and RAM to store it.
2. The flow is generally commander -> navigator but here the commander is subscribing to navigator again to check what it is doing when really navigator should just be executing as told by commander.

Alternatives I see:
1. Add this check into the mission feasibility checker. So when it checks that the mission is valid it also checks if it is set to the beginning?
2. Don't start a mission in the controller even if armed if the item is not a takeoff item.

@MaEtUgR do you have other ideas?

Reply:
@julianoes How about using the topic mission_result for this? To me this seems to be a topic which navigator uses to report various information regarding the current mission.
Wouldn't it be as simple as checking if mission_result.seq_current == 0?

Reply:
Yes, good idea!